### ISSUE TITLE
[Assetto Corsa]: Shimmering in outermost ring.

### ISSUE THREAD

**Issue by scarystuff** [fholger/openvr_foveated#5]
**Which game is affected?**
Only tested with Assetto Corsa with CSP mod so far.

**Describe the issue**
The graphic in the outermost ring seems to shimmer a bit, possibly from the sharpening effect. Might be better to leave sharpening out from that ring? Or maybe an option to define the sharpness for the different rings, if possible?

**Hardware**
Valve Index on RTX 3060Ti

**Config and log**
default cfg file.
[openvr_mod.log](https://github.com/fholger/openvr_foveated/files/7774272/openvr_mod.log)

Also, can we get an option to change the hotkey since F1 is probably used for other things already in most games. In Assetto Corsa it changes the camera view, which makes it harder to compare with the effect on and off.

Other than that is seems to work perfectly with Assetto Corsa, but I haven't really tested performance increase yet.

**Comment by scarystuff** [comment]
ok, done more testing. Turned sharpen filter off and made the rings much smaller so I can see them all. What I thought was the outermost ring, was probably only the third ring. And the shimmering has nothing to do with sharpen filter but it's just an artifact of the whole foveated effect. So all in all, it seems to work 100% as intended with Assetto Corsa!

**Comment by fholger** [comment]
Indeed, shimmering in the outer rings is quite normal due to the significantly reduced resolution. The outermost ring is rendered at 1/16th the resolution, and due to the way the masking works, filling in the missing detail can cause significant flickering between frames. I'd generally recommend to push the two outer rings as far out as possible, so that you can barely even see them (in fact, you might not even want to see the outermost ring at all). They will still save you a significant amount of pixels that don't need to be rendered :)

Great to hear that it's working in principle for Assetto. All that's left is to determine whether it actually improves performance, because unfortunately that's not guaranteed.

**Comment by scarystuff** [comment]
done a lot more testing. The mod does not work properly in vanilla Assetto Corsa without the CSP mod.

Works with CSP, but after further testing and tweaking of settings in SteamVR to get rid of the last CPU lag I kept having, I can conclude that I am getting LOWER fps with the foveated mod on :-(

So the operation was a complete success, but the patient died! ;-)
